src/Datadog.AutoInstrumentation.NativeLoader/dynamic_dispatcher.cpp

```diff
@@ -0,0 +1,333 @@
+#include "dynamic_dispatcher.h"
+
+#include <filesystem>
+#include <fstream>
+#include <unordered_map>
+
+#include "logging.h"
+#include "pal.h"
+
+#if AMD64
+
+#if _WINDOWS
+const std::string currentOsArch = "win-x64";
+#elif LINUX
+const std::string currentOsArch = "linux-x64";
+#elif MACOS
+const std::string currentOsArch = "osx-x64";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#elif X86
+
+#if _WINDOWS
+const std::string currentOsArch = "win-x86";
+#elif LINUX
+const std::string currentOsArch = "linux-x86";
+#elif MACOS
+const std::string currentOsArch = "osx-x86";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#elif ARM64
+
+#if _WINDOWS
+const std::string currentOsArch = "win-arm64";
+#elif LINUX
+const std::string currentOsArch = "linux-arm64";
+#elif MACOS
+const std::string currentOsArch = "osx-arm64";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#elif ARM
+
+#if _WINDOWS
+const std::string currentOsArch = "win-arm";
+#elif LINUX
+const std::string currentOsArch = "linux-arm";
+#elif MACOS
+const std::string currentOsArch = "osx-arm";
+#else
+#error "currentOsArch not defined."
+#endif
+
+#else
+#error "currentOsArch not defined."
+#endif
+
+namespace datadog::shared::nativeloader
+{
+
+    // ************************************************************************
+
+    //
+    // public
+    //
+
+    DynamicDispatcher::DynamicDispatcher()
+    {
+        m_continuousProfilerInstance = nullptr;
+        m_tracerInstance = nullptr;
+        m_customInstance = nullptr;
+    }
+
+    void DynamicDispatcher::LoadConfiguration(std::string configFilePath)
+    {
+        if (!std::filesystem::exists(configFilePath))
+        {
+            Warn("Configuration file doesn't exist.");
+            return;
+        }
+
+        std::ifstream t;
+        t.open(configFilePath);
```

Comment:
minor: you can pass the config file to the `std::ifstream` ctor, it will open it.

Reply:
done in 72eec52